zone/client_packet.cpp

```diff
@@ -16533,3 +16644,306 @@ void Client::ReloadExpansionProfileSetting()
 		m_pp.expansions = RuleI(World, ExpansionSettings);
 	}
 }
+
+void Client::Handle_OP_GuildTributeSelect(const EQApplicationPacket* app)
+{
+	LogTribute("Received OP_GuildSelectTribute of length [{}]", app->size);
+
+	if (app->size != sizeof(GuildTributeSelectReq_Struct))
+		LogError("Invalid size on OP_GuildSelectTribute packet");
+	else {
+		GuildTributeSelectReq_Struct* t = (GuildTributeSelectReq_Struct*)app->pBuffer;
+		SendGuildTributeDetails(t->tribute_id, t->tier);
+	}
+	return;
+}
+
+void Client::Handle_OP_GuildTributeModifyBenefits(const EQApplicationPacket* app)
+{
+	LogTribute("Received OP_Handle_OP_GuildModifyBenefits of length [{}]", app->size);
+
+	if (app->size != sizeof(GuildTributeModifyBenefits_Struct)) {
+		LogError("Invalid size on Handle_OP_GuildModifyBenefits packet");
+	}
+	else {
+		GuildTributeModifyBenefits_Struct* t = (GuildTributeModifyBenefits_Struct*)app->pBuffer;
+
+		EQApplicationPacket* outapp = new EQApplicationPacket(OP_GuildModifyBenefits, sizeof(GuildTributeModifyBenefits_Struct));
+		GuildTributeModifyBenefits_Struct* gmbs = (GuildTributeModifyBenefits_Struct*)outapp->pBuffer;
+
+		switch (t->command) {
+		case 0: {
+			gmbs->command = 0;
+			gmbs->data = 1;
+			gmbs->tribute_id_1 = t->tribute_id_1;
+			gmbs->tribute_id_2 = t->tribute_id_2;
+			gmbs->tribute_id_1_tier = t->tribute_id_1_tier;
+			gmbs->tribute_id_2_tier = t->tribute_id_2_tier;
+			gmbs->tribute_master_id = t->tribute_master_id;
+			QueuePacket(outapp);
+
+			auto sp = new ServerPacket(ServerOP_GuildTributeActivate, sizeof(GuildTributeUpdate));
+			auto out = (GuildTributeUpdate*)sp->pBuffer;
+			out->guild_id = GuildID();
+			out->enabled = false;
+			worldserver.SendPacket(sp);
+			safe_delete(sp);
+			break;
+		}
+		case 1: {
+			gmbs->command = 1;
+			gmbs->data = 1;
+			gmbs->tribute_id_1 = t->tribute_id_1;
+			gmbs->tribute_id_2 = t->tribute_id_2;
+			gmbs->tribute_id_1_tier = t->tribute_id_1_tier;
+			gmbs->tribute_id_2_tier = t->tribute_id_2_tier;
+			gmbs->tribute_master_id = t->tribute_master_id;
+			QueuePacket(outapp);
+			break;
+		}
+		default: {
+			gmbs->command = 0;
+			gmbs->data = 1;
+			gmbs->tribute_id_1 = 0;
+			gmbs->tribute_id_2 = 0;
+			gmbs->tribute_id_1_tier = 0;
+			gmbs->tribute_id_2_tier = 0;
+			gmbs->tribute_master_id = 0;
+			QueuePacket(outapp);
+			break;
+		}
+		}
+		safe_delete(outapp);
+	}
+	return;
+}
+
+void Client::Handle_OP_GuildTributeOptInOut(const EQApplicationPacket* app)
+{
+	LogTribute("Received OP_Handle_OP_GuildTributeOptInOut of length [{}]", app->size);
```

Comment:
We get this logging for free with our existing packet logging, can remove

Reply:
Removed.